iroha/test_network/src/lib.rs

```diff
@@ -207,67 +217,88 @@ impl Peer {
         configuration: Configuration,
         permissions: impl Into<PermissionsValidatorBox> + Send + 'static,
         temp_dir: &TempDir,
-    ) -> task::JoinHandle<()> {
+    ) -> Sender<ShutdownRuntime> {
```

Comment:
I guess after thread stopping it should do dropping of the runtime and everything in thread

Reply:
Yes, it will drop runtime when the thread is stopped. The thing is right now it is not a task but a runtime is spawned for each peer.

Reply:
So what is the reason then for not using JoinHandle here?

Reply:
Runtime does not have a join handle :)

Reply:
Runtime is spawned in a separate thread and as you know you can not abort thread as you can abort a task

Reply:
Oh, that makes sence, didn't know that thread couldn't be cancelled. Posix actually does have cancell function for threads.

Reply:
Such cancellation will not run all the necessary Drop checks and might cancel an operation midway as it will just abort splitting at some assembly instruction I guess. So this is why I think rust does not support it.